### DISCUSSION TITLE
Support for GOPROXY during go feature install

### DISCUSSION THREAD

**Discussion by jongray** [devcontainers/community#148] [General]
Hi,

Is there currently any way to be able to inject a GOPROXY environment variable into the go feature install?

I work in an environment where proxy.golang.org is not accessible but using GOPROXY=direct works.  Is there a way to ensure this environment variable is set so that when the go feature install runs all the tools successfully install.

I have tried remoteEnv and containerEnv but they setup the environment for after the container has been setup so aren't appropriate.

Thanks,
Jon

**Comment by samruddhikhandale** [comment]
Hi 👋 

Can you try using the Feature as follows and see if that works? 

```jsonc
"features": {
    "ghcr.io/devcontainers/features/go:1": {
        "goProxy": "direct"
    }
}

```

Even though `goProxy` is not a valid Feature option, it is still passed as the `GOPROXY` environment variable to the Feature.

If this still doesn't work, can you open an issue at https://github.com/devcontainers/features/issues, and we can look into adding it as an official Feature option? Thanks!
